Netty IllegalReferenceCountException

**Describe the bug**
We are sending post requests using the https client to a server. The requests fail with a netty error.
I am using the version 1.0.0.0-RC24

From the netty logs:

```
io.netty.util.IllegalReferenceCountException: refCnt: 0, decrement: 1
	at io.netty.util.internal.ReferenceCountUpdater.toLiveRealRefCnt(ReferenceCountUpdater.java:74)
	at io.netty.util.internal.ReferenceCountUpdater.release(ReferenceCountUpdater.java:138)
	at io.netty.buffer.AbstractReferenceCountedByteBuf.release(AbstractReferenceCountedByteBuf.java:100)
	at io.netty.util.ReferenceCountUtil.release(ReferenceCountUtil.java:90)
	at io.netty.util.ReferenceCountUtil.safeRelease(ReferenceCountUtil.java:116)
	at io.netty.channel.AbstractCoalescingBufferQueue.remove(AbstractCoalescingBufferQueue.java:181)
	at io.netty.handler.ssl.SslHandler.wrap(SslHandler.java:817)
	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1439)
	at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1235)
	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1284)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:510)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:449)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:279)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.channel.kqueue.AbstractKQueueStreamChannel$KQueueStreamUnsafe.readReady(AbstractKQueueStreamChannel.java:544)
	at io.netty.channel.kqueue.AbstractKQueueChannel$AbstractKQueueUnsafe.readReady(AbstractKQueueChannel.java:383)
	at io.netty.channel.kqueue.KQueueEventLoop.processReady(KQueueEventLoop.java:211)
	at io.netty.channel.kqueue.KQueueEventLoop.run(KQueueEventLoop.java:289)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
```

From our logs:
```
Fiber failed.
A checked error was not handled.
io.netty.handler.codec.DecoderException: io.netty.util.IllegalReferenceCountException: refCnt: 0
\tat io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:480)
\tat io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:279)
\tat io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
\tat io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
\tat io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
\tat io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
\tat io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
\tat io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
\tat io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
\tat io.netty.channel.kqueue.AbstractKQueueStreamChannel$KQueueStreamUnsafe.readReady(AbstractKQueueStreamChannel.java:544)
\tat io.netty.channel.kqueue.AbstractKQueueChannel$AbstractKQueueUnsafe.readReady(AbstractKQueueChannel.java:383)
\tat io.netty.channel.kqueue.KQueueEventLoop.processReady(KQueueEventLoop.java:211)
\tat io.netty.channel.kqueue.KQueueEventLoop.run(KQueueEventLoop.java:289)
\tat io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
\tat io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
\tat io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
\tat java.lang.Thread.run(Thread.java:748)
Caused by: io.netty.util.IllegalReferenceCountException: refCnt: 0
\tat io.netty.buffer.AbstractByteBuf.ensureAccessible(AbstractByteBuf.java:1454)
\tat io.netty.buffer.AbstractByteBuf.checkIndex(AbstractByteBuf.java:1383)
\tat io.netty.buffer.UnpooledHeapByteBuf.internalNioBuffer(UnpooledHeapByteBuf.java:316)
\tat io.netty.handler.ssl.SslHandler.wrap(SslHandler.java:1024)
\tat io.netty.handler.ssl.SslHandler.wrap(SslHandler.java:827)
\tat io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1439)
\tat io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1235)
\tat io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1284)
\tat io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:510)
\tat io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:449)
\t... 16 more

Fiber:Id(1644487685001,1) was supposed to continue to:
  a future continuation at zio.ZIO.onInterrupt(ZIO.scala:1076)
  a future continuation at zhttp.service.Client.request(Client.scala:37)
  a future continuation at zhttp.service.Client$.request(Client.scala:152)
  a future continuation at zhttp.service.Client$.request(Client.scala:144)

```
**To Reproduce**
 I am unable to reproduce this bug on my local. However it happens very quickly on our production environment.

--- a/zio-http/src/main/scala/zhttp/service/client/ClientInboundHandler.scala
+++ b/zio-http/src/main/scala/zhttp/service/client/ClientInboundHandler.scala
@@ -21,7 +21,7 @@ final class ClientInboundHandler[R](
       ctx.fireChannelActive(): Unit
     } else {
       ctx.writeAndFlush(jReq)
-      releaseRequest()
+      ()
     }
   }
 

--- a/zio-http/src/test/scala/zhttp/service/SSLSpec.scala
+++ b/zio-http/src/test/scala/zhttp/service/SSLSpec.scala
@@ -10,7 +10,7 @@ import zio.ZIO
 import zio.duration.durationInt
 import zio.test.Assertion.equalTo
 import zio.test.TestAspect.{ignore, timeout}
-import zio.test.{DefaultRunnableSpec, assertM}
+import zio.test.{DefaultRunnableSpec, Gen, assertM, checkAllM}
 
 object SSLSpec extends DefaultRunnableSpec {
   val env = EventLoopGroup.auto() ++ ChannelFactory.auto ++ ServerChannelFactory.auto
@@ -24,8 +24,15 @@ object SSLSpec extends DefaultRunnableSpec {
   val clientSSL2 =
     SslContextBuilder.forClient().trustManager(getClass().getClassLoader().getResourceAsStream("ss2.crt.pem")).build()
 
-  val app: HttpApp[Any, Nothing] = Http.collectZIO[Request] { case Method.GET -> !! / "success" =>
-    ZIO.succeed(Response.ok)
+  val payload = Gen.alphaNumericStringBounded(10000, 20000)
+
+  val app: HttpApp[Any, Throwable] = Http.collectZIO[Request] {
+    case Method.GET -> !! / "success"     =>
+      ZIO.succeed(Response.ok)
+    case req @ Method.POST -> !! / "text" =>
+      for {
+        body <- req.bodyAsString
+      } yield Response.text(body)
   }
 
   override def spec = suiteM("SSL")(
@@ -59,7 +66,20 @@ object SSLSpec extends DefaultRunnableSpec {
                 .request("http://localhost:8073/success", ssl = ClientSSLOptions.CustomSSL(clientSSL1))
                 .map(_.status)
               assertM(actual)(equalTo(Status.PERMANENT_REDIRECT))
-            } @@ ignore,
+            } +
+            testM("Https request with a large payload should respond with 413") {
+              checkAllM(payload) { payload =>
+                val actual = Client
+                  .request(
+                    "https://localhost:8073/text",
+                    Method.POST,
+                    ssl = ClientSSLOptions.CustomSSL(clientSSL1),
+                    content = HttpData.fromString(payload),
+                  )
+                  .map(_.status)
+                assertM(actual)(equalTo(Status.REQUEST_ENTITY_TOO_LARGE))
+              }
+            },
         ),
       )
       .useNow,

